Lib/test/test_types.py

```diff
@@ -772,6 +775,31 @@ def test_union_parameter_chaining(self):
         self.assertEqual((list[T] | list[S])[int, T], list[int] | list[T])
         self.assertEqual((list[T] | list[S])[int, int], list[int])
 
+    def test_union_pickle(self):
+        alias = list[T] | int
+        s = pickle.dumps(alias)
+        loaded = pickle.loads(s)
+        self.assertEqual(alias.__args__, loaded.__args__)
+        self.assertEqual(alias.__parameters__, loaded.__parameters__)
+
+    def test_union_from_args(self):
+        with self.assertRaisesRegex(
+                TypeError,
+                r"^Each union argument must be a type, got 1$",
+        ):
+            types.Union.from_args((1,))
+
+        with self.assertRaisesRegex(
+                TypeError,
+                r"Union.from_args\(\) argument '__args__' must be tuple, not int$",
+        ):
+            types.Union.from_args(1)
+
+        alias = types.Union.from_args((int, str, T))
```

Comment:
At first I was a little weirded out by the single argument that is a tuple but looking at actual usage in `typing.py` this is the right choice 👍🏻

Using literals like in this test will be a rare occurence because then you can use the regular syntax for the union.

Reply:
Agree, I used `from_args` only to test behaviour of this classmethod.